### ISSUE TITLE
[Train] Add callback preprocessor that smoothly tracks values 

### ISSUE THREAD

**Issue by bveeramani** [ray-project/ray#21989]
### Search before asking

- [X] I had searched in the [issues](https://github.com/ray-project/ray/issues) and found no similar feature requirement.

### Description

The [reference classification](https://github.com/pytorch/vision/blob/main/references/classification/utils.py) script for PyTorch vision implements a class that smoothly tracks distributed training data across epochs (e.g., loss).

```
class SmoothedValue:
    """Track a series of values and provide access to smoothed values over a
    window or the global series average.
    """
    ...
```

We could add a callback or callback preprocessor that provides similar functionality.

### Use case

This feature would make it easier for users to log training data like accuracy and loss.

### Related issues

#21367 implements preprocessing functionality.

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

**Comment by cszhu** [comment]
Hello! This P2 issue has seen no activity in at least 2 years. It will be closed in 2 weeks as part of ongoing cleanup efforts.

Please remove the `pending-cleanup` label if you believe this issue should remain open.

Thanks for contributing to Ray!
